### ISSUE TITLE
Lightswitch detection - weird values in the logs

### ISSUE THREAD

**Issue by lmarceg** [Motion-Project/motion#1079]
Hi,
I am playing a bit with lightswitch but it is very difficult for me to find the right values.
In the settings, my camera has a width of 640 and a height of 352. Lightswitch is set to 18%.
Now, according to the code, `cnt->imgs.motionsize = cnt->conf.width * cnt->conf.height;` and lightswitch is detected `if (diffs > (imgs->motionsize * cnt->conf.lightswitch_percent / 100))`
In my case, motionsize=640*352=225280, so diffs should be >40550 which is 18% of motionsize.
Instead, I get a Lightswitch notification for the following data:

```
[1:ml1:Terrazza] [INF] [ALL] mlp_setupmode: Raw changes:  8487 - changes after 'EedDl':  2243 - labels:  11 - noise level: 18
[1:ml1:Terrazza] [INF] [ALL] mlp_setupmode: Raw changes:  8487 - changes after 'EedDl':  2243 - labels:  11 - noise level: 18
[1:ml1:Terrazza]ü[INF]ü[ALL]üAbove message repeats 1 times
[1:ml1:Terrazza]ü[INF]ü[ALL]ümlp_setupmode: Raw changes:  8487 - changes after 'EedDl':  2243 - labels:  11 - noise level: 18
[1:ml1:Terrazza]ü[INF]ü[ALL]ümlp_setupmode: Raw changes:  8487 - changes after 'EedDl':  2243 - labels:  11 - noise level: 18
[1:ml1:Terrazza]ü[INF]ü[ALL]ümlp_setupmode: Raw changes:  8487 - changes after 'EedDl':  2243 - labels:  11 - noise level: 18
[1:ml1:Terrazza]ü[INF]ü[ALL]ümlp_detection: Lightswitch detected
```

I get similar detection also for lower values, though it should work just the opposite: I should get a detection when the Raw Changes (or the changed after EedDl ?) are >40.000....
Where am I wrong? The problem is that I need to experiment manually each time and lightswitch depends on the ToD, as external light changes, impacting the overall brightness of the scene when the light is turned on (so basically, the % value is almost impossible to set correctly). On top, I don't get any % in the logs and I need to do manual calculations (provided mine above are right).
I would also like to add a timestamp to the logs but couldn't find any way to. This would ease my debugging.

Thanks for helping!
Luca

**Comment by tosiara** [comment]
Try to use syslog for logging. That way you will see timestamps

Could you reproduce the issue and provide a detailed log with `-d 9` for a lightswitch with a corresponding debug video?

**Comment by lmarceg** [comment]
Thanks for the timestamp suggestion. I studied the coded and decided to manually run motion with -s, a log file, and with -d 9 so I found out different results which explained better the way lightswitch is detected. Motion is just one factor, being luminance the other one the algorithm is taking into account.
This explains while I could see low motion values but possibly luminance change (unfortunately, these values are not shown in the logs).
I am closing this case for now.
Thanks
